src/gennylib/src/context.cpp

```diff
@@ -58,9 +59,17 @@ WorkloadContext::WorkloadContext(YAML::Node node,
 
     _clientPool = poolFactory.makePool();
 
+    // TODO: Remove hasApmOpts when CDRIVER-2931 is resolved.
+    auto apmOpts = clientOpts.apm_opts();
```

Comment:
I'm confused - when do we actually set `clientOpts` and when does it get passed in? The driver is the only thing that creates the `WorkloadContext` and that's not being changed here from what I see so it seems like the `apm_opts()` will always just be whatever it is from the default ctor, ya?

Reply:
The clientOpts are being set with the apm options here: https://github.com/mongodb/genny/blob/733e2991c30f8215c07d60a952cc502f02222e67/src/gennylib/test/CrudActor_test.cpp#L27-L39. A `SessionTest` object is initialized in the necessary test cases and the client options are then passed in the constructor of ActorHelper.

Reply:
do we need to do the same thing in the driver code then? not sure why tests would be different? or since we're only intending the user to pass in a callback, maybe the API should take the callback as part of the ctor and let the PoolFactory construct everything? just throwing out ideas